Cannot build on Apple M1

**Describe the bug**
Getting compiler errors when trying to build on Apple M1

**To Reproduce**
Steps to reproduce the behavior:
1. clone the reporitory
2. run `make app`

**Software involved**
Please complete the following information:
 - OS: MacOS 11.2.3
 - Ajour version: latest master (varsion 1.0.1+) 


**Log Output**
```
cargo build --release
   Compiling winit v0.23.0
   Compiling ajour-core v1.0.1 (/Users/gl/Projects/ajour/crates/core)
   Compiling ajour-widgets v1.0.1 (/Users/gl/Projects/ajour/crates/widgets)
error[E0308]: mismatched types
   --> /Users/gl/.cargo/registry/src/github.com-1ecc6299db9ec823/winit-0.23.0/src/platform_impl/macos/util/async.rs:170:55
    |
170 |                 ns_window.setFrame_display_(new_rect, 0);
    |                                                       ^ expected `bool`, found integer

error[E0308]: mismatched types
   --> /Users/gl/.cargo/registry/src/github.com-1ecc6299db9ec823/winit-0.23.0/src/platform_impl/macos/view.rs:391:9
    |
386 | extern "C" fn has_marked_text(this: &Object, _sel: Sel) -> BOOL {
    |                                                            ---- expected `bool` because of return type
...
391 |         (marked_text.length() > 0) as i8
    |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected `bool`, found `i8`

error[E0308]: mismatched types
   --> /Users/gl/.cargo/registry/src/github.com-1ecc6299db9ec823/winit-0.23.0/src/platform_impl/macos/window.rs:656:22
    |
656 |         is_zoomed != 0
    |                      ^ expected `bool`, found integer

error[E0308]: mismatched types
    --> /Users/gl/.cargo/registry/src/github.com-1ecc6299db9ec823/winit-0.23.0/src/platform_impl/macos/window.rs:1171:48
     |
1171 |         window.setFrame_display_(current_rect, 0)
     |                                                ^ expected `bool`, found integer

error[E0308]: mismatched types
    --> /Users/gl/.cargo/registry/src/github.com-1ecc6299db9ec823/winit-0.23.0/src/platform_impl/macos/window.rs:1178:48
     |
1178 |         window.setFrame_display_(current_rect, 0)
     |                                                ^ expected `bool`, found integer

error[E0308]: mismatched types
    --> /Users/gl/.cargo/registry/src/github.com-1ecc6299db9ec823/winit-0.23.0/src/platform_impl/macos/window.rs:1195:48
     |
1195 |         window.setFrame_display_(current_rect, 0)
     |                                                ^ expected `bool`, found integer

error[E0308]: mismatched types
    --> /Users/gl/.cargo/registry/src/github.com-1ecc6299db9ec823/winit-0.23.0/src/platform_impl/macos/window.rs:1202:48
     |
1202 |         window.setFrame_display_(current_rect, 0)
     |                                                ^ expected `bool`, found integer

error: aborting due to 7 previous errors

For more information about this error, try `rustc --explain E0308`.
error: could not compile `winit`

To learn more, run the command again with --verbose.
warning: build failed, waiting for other jobs to finish...
error: build failed
make: *** [ajour] Error 101
```

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -21,9 +21,9 @@ ajour-core = { version = "1.1.0-rc.1", path = "crates/core", features=['gui'] }
 ajour-weak-auras = { version = "1.1.0-rc.1", path = "crates/weak_auras" }
 ajour-widgets = { version = "1.1.0-rc.1", path = "crates/widgets" }
 
-iced = { version = "0.2.0", default-features = false, features = ["debug"] }
-iced_futures = { version = "0.2", features = ["async-std"] }
-iced_native = { version = "0.3" }
+iced = { version = "0.3.0", default-features = false, features = ["debug"] }
+iced_futures = { version = "0.3.0", features = ["async-std"] }
+iced_native = { version = "0.4.0" }
 async-std = "1.6.2"
 isahc = { version = "0.9.6", features = ["json"] }
 image = "0.23.8"

--- a/crates/core/Cargo.toml
+++ b/crates/core/Cargo.toml
@@ -48,7 +48,7 @@ tar = "0.4.33"
 zstd = { version = "0.6.1", features = ["zstdmt"] }
 num_cpus = "1.13.0"
 
-iced_native = { version = "0.3", optional = true }
+iced_native = { version = "0.4.0", optional = true }
 
 [target.'cfg(target_os = "macos")'.dependencies]
 flate2 = "1.0"

--- a/crates/widgets/Cargo.toml
+++ b/crates/widgets/Cargo.toml
@@ -14,9 +14,9 @@ wgpu = ["iced_wgpu"]
 opengl = ["iced_glow"]
 
 [dependencies]
-iced_native = { version = "0.3" }
-iced_graphics = { version = "0.1" }
-iced_core = { version = "0.3" }
+iced_native = { version = "0.4.0" }
+iced_graphics = { version = "0.2.0" }
+iced_core = { version = "0.4.0" }
 
-iced_wgpu = { version = "0.3", optional = true }
-iced_glow = { version = "0.1", optional = true }
+iced_wgpu = { version = "0.4.0", optional = true }
+iced_glow = { version = "0.2.0", optional = true }

--- a/crates/widgets/src/widget/header.rs
+++ b/crates/widgets/src/widget/header.rs
@@ -152,9 +152,9 @@ where
         event: Event,
         layout: Layout<'_>,
         cursor_position: Point,
-        messages: &mut Vec<Message>,
         renderer: &Renderer,
-        clipboard: Option<&dyn Clipboard>,
+        clipboard: &mut dyn Clipboard,
+        messages: &mut Vec<Message>,
     ) -> event::Status {
         let in_bounds = layout.bounds().contains(cursor_position);
 
@@ -223,9 +223,9 @@ where
                         return event::Status::Captured;
                     }
                 }
-                Event::Mouse(mouse::Event::CursorMoved { x, .. }) => {
+                Event::Mouse(mouse::Event::CursorMoved { position }) => {
                     if self.state.resizing {
-                        let delta = x - self.state.starting_cursor_pos.unwrap().x;
+                        let delta = position.x - self.state.starting_cursor_pos.unwrap().x;
 
                         let left_width = self.state.starting_left_width;
                         let right_width = self.state.starting_right_width;
@@ -261,9 +261,9 @@ where
                     event.clone(),
                     layout,
                     cursor_position,
-                    messages,
                     renderer,
                     clipboard,
+                    messages,
                 )
             })
             .fold(event::Status::Ignored, event::Status::merge)

--- a/crates/widgets/src/widget/table_row.rs
+++ b/crates/widgets/src/widget/table_row.rs
@@ -193,17 +193,17 @@ where
         event: Event,
         layout: Layout<'_>,
         cursor_position: Point,
-        messages: &mut Vec<Message>,
         renderer: &Renderer,
-        clipboard: Option<&dyn Clipboard>,
+        clipboard: &mut dyn Clipboard,
+        messages: &mut Vec<Message>,
     ) -> event::Status {
         let status_from_content = self.content.on_event(
             event.clone(),
             layout.children().next().unwrap(),
             cursor_position,
-            messages,
             renderer,
             clipboard,
+            messages,
         );
         match status_from_content {
             event::Status::Ignored => {

--- a/src/gui/mod.rs
+++ b/src/gui/mod.rs
@@ -26,9 +26,9 @@ use ajour_widgets::header;
 use async_std::sync::{Arc, Mutex};
 use chrono::{DateTime, NaiveDateTime, Utc};
 use iced::{
-    button, pick_list, scrollable, text_input, Align, Application, Button, Column, Command,
-    Container, Element, HorizontalAlignment, Length, PickList, Row, Scrollable, Settings, Space,
-    Subscription, Text, TextInput,
+    button, pick_list, scrollable, text_input, Align, Application, Button, Clipboard, Column,
+    Command, Container, Element, HorizontalAlignment, Length, PickList, Row, Scrollable, Settings,
+    Space, Subscription, Text, TextInput,
 };
 use image::ImageFormat;
 use isahc::http::Uri;
@@ -341,7 +341,7 @@ impl Application for Ajour {
         ])
     }
 
-    fn update(&mut self, message: Message) -> Command<Message> {
+    fn update(&mut self, message: Message, _clipboard: &mut Clipboard) -> Command<Message> {
         match update::handle_message(self, message) {
             Ok(x) => x,
             Err(e) => Command::perform(async { e }, Message::Error),

